jetty-core/jetty-http/src/test/java/org/eclipse/jetty/http/MultiPartCaptureTest.java

```diff
@@ -244,9 +244,8 @@ private void assertParts(Map<String, List<MultiPart.Part>> allParts) throws Exce
             List<MultiPart.Part> charSetParts = allParts.get("_charset_");
             if (charSetParts != null)
             {
-                Promise.Completable<String> promise = new Promise.Completable<>();
-                Content.Source.asString(charSetParts.get(0).getContent(), StandardCharsets.US_ASCII, promise);
-                defaultCharset = promise.get();
+                defaultCharset = Promise.Completable.<String>with(p -> Content.Source.asString(charSetParts.get(0).getContent(), StandardCharsets.US_ASCII, p))
```

Comment:
Too bad we can't have `Content.Source.asString(Source, Charset)` that returns a `CompletableFuture`.